modules/site-to-site-vpn/variables.tf

```diff
@@ -0,0 +1,120 @@
+    - `bgp_asn`: The Border Gateway Protocol (BGP) Autonomous System Number (ASN) required for BGP routing between the VPN and the customer gateway.
+    - `certificate_arn`: When required, the ARN of the customer gateway certificate for certificate-based VPN authentication.
+    - `device_name`: An optional field for naming or identifying the customer gateway device in your infrastructure.
+    - `ip_address`: The public-facing IP address that AWS will use to connect to the customer gateway.
+    - `type`: Specifies the type of VPN connection, defaulting to 'ipsec.1' for most use cases.
+  EOT
+}
+
+variable "vpn_gateway_config" {
+  type = object({
+    create            = optional(bool, true)
+    vpc_id            = string                 # The VPC ID to create the VPN gateway in.
+    amazon_side_asn   = optional(number, null) # The Autonomous System Number (ASN) for the Amazon side of the gateway.
+    availability_zone = optional(string, null) # The Availability Zone for the VPN gateway.
+  })
+  description = <<-EOT
+    Configuration for creating an AWS VPN Gateway. This object allows you to define:
+    - `vpc_id`: The VPC ID where the VPN Gateway will be created.
+    - `amazon_side_asn`: Optional ASN for the Amazon side of the VPN gateway.
+    - `availability_zone`: Optional specification of an availability zone for the VPN Gateway.
+  EOT
+}
+
+variable "vpn_connection_config" {
+  type = object({
+    transit_gateway_id  = optional(string, null) # The ID of the transit gateway
+    static_routes_only  = optional(bool, false)  # If true, only static routes are used
+    enable_acceleration = optional(bool, false)  # (Optional, Default false) Indicate whether to enable acceleration for the VPN connection. Supports only EC2 Transit Gateway.
+
+    # New fields for local and remote CIDR blocks, outside IP address type, and transit gateway attachment
+    local_ipv4_network_cidr                 = optional(string, "0.0.0.0/0")  # The IPv4 CIDR on the customer gateway side
+    local_ipv6_network_cidr                 = optional(string, "::/0")       # The IPv6 CIDR on the customer gateway side
+    outside_ip_address_type                 = optional(string, "PublicIpv4") # Public or Private S2S VPN
+    remote_ipv4_network_cidr                = optional(string, "0.0.0.0/0")  # The IPv4 CIDR on the AWS side
+    remote_ipv6_network_cidr                = optional(string, "::/0")       # The IPv6 CIDR on the AWS side
+    transport_transit_gateway_attachment_id = optional(string, null)         # Transit Gateway attachment ID (required for PrivateIpv4)
+
+    # Tunnel configuration options
+    tunnel_config = object({
+      tunnel1 = object({
+        inside_cidr                  = string                                       # CIDR block of the first tunnel
+        preshared_key                = optional(string, null)                       # Pre-shared key for the first tunnel
```

Comment:
The preshared_key is sensitive. Will show up in plan and output if not sensitized.

```
variable "tunnel1_preshared_key" {
  type      = string
  sensitive = true
}

variable "tunnel2_preshared_key" {
  type      = string
  sensitive = true
}
```

Reply:
Thanks, updated